src/main/java/org/variantsync/diffdetective/diff/difftree/parse/DiffTreeParser.java

```diff
@@ -102,35 +96,24 @@ public static DiffResult<DiffTree> createDiffTree(
             }
 
             // check if this is a multiline macro
-            final ParseResult isMLMacro;
             try {
-                isMLMacro = mlMacroParser.consume(lineNo, currentLine, beforeStack, afterStack);
-            } catch (IllFormedAnnotationException e) {
-                return DiffResult.Failure(e);
-            }
-
-            switch (isMLMacro.type()) {
-                case Success: {
+                if (mlMacroParser.consume(lineNo, currentLine, beforeStack, afterStack)) {
                     if (lastArtifact != null) {
                         lastArtifact = endCodeBlock(lastArtifact, lastLineNo);
                     }
                     // This line belongs to a multiline macro and was handled, so go to the next line.
                     continue;
                 }
-                case Error: {
-                    isMLMacro.onError(errorPropagation);
-                    return error.get();
-                }
-                // line is not a mult-line macro so keep going (break the switch statement).
-                case NotMyDuty: break;
+            } catch (IllFormedAnnotationException e) {
+                throw new DiffParseException(e.getType(), lineNo);
             }
+            // line is not a mult-line macro so keep going (break the switch statement).
```

Comment:
The comment should be adapted because `(break the switch statement)` does not apply anymore.

Reply:
I just removed the part in the brackets. This comment is also removed in a later commit, so I mark this as resolved.